File: manual/data/cpp/src/function_itertools_cycle.cpp
#include <string>
#include <boost/iterator/iterator_facade.hpp>

class CycleProcessor
{
public:
    std::string test_cycle(const std::string &input_sequence, int num_elements)
    {
        if (input_sequence.empty())
        {
            return "";
        }

        std::string result;
        result.reserve(num_elements);

        size_t input_length = input_sequence.length();
        for (int i = 0; i < num_elements; ++i)
        {
            result += input_sequence[i % input_length];
        }

        return result;
    }
};

File: manual/code/generate_strategies/IR(pseudocode)/python/python/function_itertools_cycle.py
import itertools


class CycleProcessor:
    def test_cycle(self, input_sequence: str, num_elements: int) -> str:
        if not input_sequence:
            return ''

        cycle_iter = itertools.cycle(input_sequence)
        result = ""
        for _ in range(num_elements):
            result += next(cycle_iter)
        return result
